src/VisualStudio/TestUtilities/IntegrationHost.cs

```diff
@@ -0,0 +1,326 @@
+                    {
+                        projectFiles.Add(project.FullName);
+                    }
+
+                    // Save the full path to the solution. This is so we can cleanup any folders after the solution is closed.
+                    var solutionFile = _dte.Solution.FullName;
+
+                    _dte.Solution.Close(SaveFirst: false);
+
+                    foreach (var projectFile in projectFiles)
+                    {
+                        var projectFolder = Path.GetDirectoryName(projectFile);
+                        _requireNewInstance |= IntegrationHelper.TryDeleteDirectoryRecursively(projectFolder);
+                    }
+
+                    if (!string.IsNullOrWhiteSpace(solutionFile))
+                    {
+                        var solutionFolder = Path.GetDirectoryName(solutionFile);
+                        _requireNewInstance |= IntegrationHelper.TryDeleteDirectoryRecursively(solutionFolder);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.WriteLine($"Warning: Failed to cleanup the DTE.");
+                Debug.WriteLine($"\t{e}");
+                _requireNewInstance |= true;
+            }
+        }
+
+        private void CleanupHostProcess()
+        {
+            if (_dte != null)
+            {
+                _dte.Quit();
+                _dte = null;
+            }
+
+            IntegrationHelper.KillProcess(_hostProcess);
+            _hostProcess = null;
+        }
+
+        private void CleanupRemotingService()
+        {
+            try
+            {
+                if ((_dte?.Commands.Item(VsStopServiceCommand).IsAvailable).GetValueOrDefault())
+                {
+                    ExecuteDteCommandAsync(VsStopServiceCommand).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+                }
+            }
+            finally
+            {
+                _service = null;
+                _serviceUri = null;
+
+                if (_serviceChannel != null)
+                {
+                    ChannelServices.UnregisterChannel(_serviceChannel);
+                    _serviceChannel = null;
+                }
+            }
+        }
+
+        private void InitializeDte()
+            => IntegrationHelper.WaitForResultAsync(() => {
+                _dte = IntegrationHelper.LocateDteForProcess(_hostProcess);
+                return (_dte != null);
+            }, expectedResult: true).ConfigureAwait(continueOnCapturedContext: false).GetAwaiter().GetResult();
+
+        private void InitializeHostProcess()
+        {
+            Process.Start(VsExeFile, $"/clearcache {VsLaunchArgs}").WaitForExit();
+            Process.Start(VsExeFile, $"/updateconfiguration {VsLaunchArgs}").WaitForExit();
+
+            IntegrationHelper.KillProcess("DbgCLR");
+            IntegrationHelper.KillProcess("VsJITDebugger");
+            IntegrationHelper.KillProcess("dexplore");
```

Comment:
Our internal StartTargetAction did this, I believe it was to ensure that no handles were left hanging on things, and because we did evil things to the AssemblyResolver which broke the MEF composition (these shouldn't be an issue anymore, but I haven't validated that to be true yet).